Clean up legacy styles and remove bs3-compat.scss

EreMaijala opened the pull request:
This is collection of sort-of independent commits with the aim to remove the Bootstrap 3 compatibility layer. 

Notable changes:

- brand- prefix for main colors no longer exists, so e.g. $brand-primary is just $primary.
- "default" is now added as a color to Bootstrap maps so that `btn btn-default` continues to work.
- All select fields that used to have `form-control` and `form-select` classes now only have `form-select`.
- `sr-only` is now `visually-hidden`
- `sr-only-focusable` is now `visually-hidden-focusable`
- `control-label` is now `form-label`
- `pull-right` is now `float-end` (`flip` no longer needed)
- `pull-left` is now `float-start` (`flip` no longer needed)
- Many renamings of variables in Bootstrap 5 have been applied to scss files throughout. Typical type of change is `$something-border` renamed to `$something-border-color`.
- Odd dotted button outline is no longer odd or dotted

Things to check during testing:

- Cookie consent dialog and preferences
- Forms and their layout (e.g. advanced search, feedback)
- Record tags
- Button colors
- Colors in general
- Badge (e.g. counts in facets and "Your Lists")
- Search box and virtual keyboard button
- Lightbox close button
- RTL languages

TODO:
- [x] Add change log entry when this is merged.
- [x] Resolve VUFIND-1743 when this is merged.

EreMaijala committed:
Replace $brand-something with just $something

EreMaijala committed:
Eliminate border-radius variables and breadcrumb-separator

EreMaijala committed:
Adopt $nav-link-hover-bg and drop other unused nav-link variables.

EreMaijala committed:
Drop unused variables.

EreMaijala committed:
Eliminate padding and panel vars.

EreMaijala committed:
Eliminate input-height-base, list-group-border and double gray-*.

EreMaijala committed:
Adopt btn-default.

EreMaijala committed:
Remove copied color variables and out-of-date components/variables.scss.

EreMaijala committed:
Adopt pager.

EreMaijala committed:
Adopt legend changes, eliminate miscellaneous variables.

EreMaijala committed:
Drop unused mixins.

EreMaijala committed:
Replace sr-only with visually-hidden and sr-only-focusable with visually-hidden-focusable.

EreMaijala committed:
Eliminate button back-compatibility styles.

EreMaijala committed:
Use Bootstrap's btn-group.

EreMaijala committed:
Adopt badges and labels.

EreMaijala committed:
Use VuFind's dropdown caret everywhere.

EreMaijala committed:
Clean up form styles

- form-select is now alone without form-control
- Some margins adjusted and feedback admin form cleaned up

EreMaijala committed:
Change control-label class to form-label.

EreMaijala committed:
Adopt form-group and input-group styles.

EreMaijala committed:
Rearrange miscellaneous rules, remove duplicates, fix textarea height.

EreMaijala committed:
Misc rearrangements, replace pull-right with float-end and pull-left with float-start.

EreMaijala commented:
@sturkel89 I hope you could check this set of changes. This is the final big change to eliminate legacy styles (I hope!). Sorry for it being so large, but thought it would be better to keep all these together because many of changes touch same areas in the UI.

sturkel89 reviewed (changes requested):
I tested this PR in both themes and in the mobile emulator, comparing test and dev. (I only tested on Chrome.) I found a number of little quirks that should be pretty easy to fix. 

---------------------

### Item record top menu: there's no down arrow in Export Record (both themes)

The down arrow is missing, but the functionality is still there; if you click the button, the three export options appear.

![image](https://github.com/user-attachments/assets/d759f065-e07b-4b70-abe2-7b962c552901)

**UPDATE: This is fixed!**

--------------------

### More button in top Suggested Topics area looks weird (both themes)

On dev, the **More** button is rounded with a faint, clean outline. 

On test, the **More** button is squared off, short, and dark outlining is present whether the button is clicked or not. Both the **More** and **Less** buttons in this feature have the same appearance.

![image](https://github.com/user-attachments/assets/7af17109-435b-43f4-90f0-6b5f81283cf4)

**UPDATE: This is fixed, but looks lighter on test than on dev.**

---------------------------

### Sidebar facet listings look weird (both themes)

On test, in bootstrap5, there's a gap between the header field and the list of filters, and the **More** button with the arrow looks weird (similar to the **More** button in **Suggested Topics**).

![image](https://github.com/user-attachments/assets/b77b5315-e739-4678-9f44-caa1c451d8c1)

In sandal5, the gap is present, but the **More** button looks okay.

![image](https://github.com/user-attachments/assets/decbe818-3dd6-4a74-8e5d-3f4b865102c3)

**UPDATE: This is fixed!**

---------------------

### Text Area boxes are three lines high on test but one line high on dev (both themes)

We think that this is an improvement and a reversion to how things used to be! So, this is more of a note than a request to change things. 

**Example: Comment box on Bootstrap5**

![image](https://github.com/user-attachments/assets/846bd3b7-5485-4603-a039-884719fe30fd)

**Example: Create a List on mobile, Sandal5** 

(NB: there are two other tiny issues with this screen: inconsistent punctuation in the labels for the two boxes, and something is weird with spacing of the breadcrumbs. I'll keep track of those in a different document.)

![image](https://github.com/user-attachments/assets/38ccf6ad-b4c8-476c-8be3-41dcaceb7420)

**UPDATE: This is fixed!**

-----------------------------

### Record tags: alignment/spacing is different between test and dev

In both themes, there's more horizontal padding inside tag buttons. I don't have a strong opinion on whether this is an improvement or not.

In Sandal5, the padding is the main difference in terms of tag appearance and behavior; wrapping behavior is pretty similar between test and dev. 

In Bootstrap5, wrapping behavior is pretty different between test and dev.

Bootstrap5: **in test**, three short tags need a very wide browser to display all in one row. This is the minimal width for having all three on one line; any narrower, and they wrap to two rows.

![image](https://github.com/user-attachments/assets/81b2979c-2654-40fd-87b1-3098893519ab)

**In dev**, this is the width that's needed to display them all in one row:
![image](https://github.com/user-attachments/assets/f1a5a635-d1af-4e2f-a1d7-95c8348ef8fe)

Here's the mobile view of the same three tags in bootstrap5:

![image](https://github.com/user-attachments/assets/00a72e71-d771-418b-8f10-d5e7a857ccbc)

**UPDATE: This is fixed!**

demiankatz reviewed (changes requested):
Looks like the merge of #4238 has introduced some conflicts. Sorry about that!

EreMaijala commented:
@demiankatz No problem, that was to be expected!

EreMaijala committed:
Fix Export Record caret.

EreMaijala committed:
Fix more/less buttons.

Now unified with toolbar-btn which was also tweaked a bit with a border and improved contrast.

EreMaijala committed:
Fix facet title margin.

EreMaijala committed:
Use smaller padding for tags.

EreMaijala commented:
@sturkel89 Thanks for the comments! All should be fixed now. Remarks:

- The one-line text areas were actually a bug. E.g. on the feedback form the HTML defined eight visible rows, but an invalid css rule overrode that.
- I dropped the ball with the More/less button. Should be better now, but intentionally not quite the same as before. This is because I took the liberty to unify them with toolbar buttons (record toolbar, bulk actions toolbar, cart lightbox toolbar) which were also tweaked a bit with slightly darker text and a subtle border to improve contrast. I did all this because I think the more/less buttons stood out a bit too much while the toolbar buttons almost blended into the background.

sturkel89 reviewed (approved):
I see that all of my requested changes have been made! Looks great.

In terms of the More/Less buttons in **Suggested Topics within your search**, I see what you're saying. If the intention is to make these More/Less buttons slightly de-emphasized, you've been successful. As long as it's not an accessibility violation to have such a subtle-looking button, I'm okay with it.

**Test:**
![image](https://github.com/user-attachments/assets/652b7d07-eff4-4b70-8a2f-478a8636fd4f)

**Dev:**
![image](https://github.com/user-attachments/assets/33d3d746-fbbe-417f-b856-2b0acbb07316)

EreMaijala commented:
@sturkel89 I verified that the contrast is high enough, and actually the toolbar buttons now have better contrast between text and background than before, because I made the text a notch darker.

demiankatz reviewed (approved):
All looks reasonable to me. Thanks, @EreMaijala and @sturkel89!

demiankatz commented:
Regarding the changelog entry: there is already a link from the changelog to the [bootstrap5 wiki page](https://vufind.org/wiki/development:architecture:user_interface:bootstrap5). I added a link from that page to this PR for reference (since there was already a link to the PR removing the JS layer, and this complements that). @EreMaijala, do you think any further documentation work is needed?

EreMaijala commented:
@demiankatz Thanks! I think that no further documentation is needed.
